adafruit_io/adafruit_io.py

```diff
@@ -40,9 +40,9 @@ def validate_feed_key(feed_key):
     """
     if len(feed_key) > 128:  # validate feed key length
         raise ValueError("Feed key must be less than 128 characters.")
-    if not bool(re.match("^[a-z0-9-]+$", feed_key)):  # validate key naming scheme
+    if not bool(re.match("^[a-z0-9-\\.]+$", feed_key)):  # validate key naming scheme
```

Comment:
@flavio-fernandes  Why did you add `\\`? IO feeds do not contain "\" characters.

Reply:
`^[a-z0-9-.]+$` is able to match a grouped feed such as `air-quality-sensor.aqi`

Reply:
Sorry for chiming in, but the regexp in the above code allows for leading/trailing dots.

Also I think that a better regexp is something like `^[a-z0-9-]+(\.[a-z0-9-]+)?$`, which matches `foo` or `foo.bar` but not `foo.` or `foo.bar.baz`. (don't know if IO feeds can have subgroups).

Reply:
@xadhoom IO feeds can't have subgroups. like that, it'll typically look like `feed.group`. Thanks for chiming in

Reply:
https://learn.adafruit.com/naming-things-in-adafruit-io/aside-naming-things-in-mqtt

Reply:
Ok, so I read it correctly. then my regexp above is more correct, because it matches only a group and blocks leading/trailing dots .

Reply:
Yes. I just did not want to make it more complex than it needs to be, following the current implementation.

Reply:
@xadhoom Please see my latest change and confirm this is what you would like the change to do.
@brentru sorry for the follow up changes. I blame @xadhoom :)

Reply:
@flavio-fernandes added a small comment to the regexp, just to be a bit more pedantic :)

Also, according to https://learn.adafruit.com/naming-things-in-adafruit-io/the-two-feed-identifiers seems that the `-` is not allowed at the beginning or at the end of the string.

So if you want a more complex one: `^[a-z0-9][a-z0-9-]*((?<!-)\.(?!-)[a-z0-9-]*)?[a-z0-9]$` . This basically avoids trailing dashes and dashes "around" the dot.

See https://regex101.com/r/5Pt0U9/1

Reply:
That is getting a little too complex. I _really_ rather not...